src/shared/bus-util.c

```diff
@@ -259,7 +259,7 @@ int bus_connect_user_systemd(sd_bus **ret_bus) {
 
         e = secure_getenv("XDG_RUNTIME_DIR");
         if (!e)
-                return -ENXIO;
+                return -ENOMEDIUM;
```

Comment:
Hmmm I still get:
```
$ systemctl --user status
Failed to connect to user scope bus via local transport: No medium found
```

Let me dig into why we're not getting proper error message

Reply:
Ah there's a bug in the log message where we compare the error code and booleans... @DaanDeMeyer can you also apply this patch:
```
--- a/src/shared/bus-util.c
+++ b/src/shared/bus-util.c
@@ -56,8 +56,8 @@ int bus_log_connect_error(int r, BusTransport transport, RuntimeScope scope) {
              hint_addr = transport == BUS_TRANSPORT_LOCAL && ERRNO_IS_PRIVILEGE(r);

         return log_error_errno(r,
-                               r == hint_vars ? "Failed to connect to %s scope bus via %s transport: $DBUS_SESSION_BUS_ADDRESS and $XDG_RUNTIME_DIR not defined (consider using --machine=<user>@.host --user to connect to bus of other user)" :
-                               r == hint_addr ? "Failed to connect to %s scope bus via %s transport: Operation not permitted (consider using --machine=<user>@.host --user to connect to bus of other user)" :
+                               hint_vars ? "Failed to connect to %s scope bus via %s transport: $DBUS_SESSION_BUS_ADDRESS and $XDG_RUNTIME_DIR not defined (consider using --machine=<user>@.host --user to connect to bus of other user)" :
+                               hint_addr ? "Failed to connect to %s scope bus via %s transport: Operation not permitted (consider using --machine=<user>@.host --user to connect to bus of other user)" :
                                                 "Failed to connect to %s scope bus via %s transport: %m", runtime_scope_to_string(scope), bus_transport_to_string(transport));
 }
```

Reply:
Yikes good catch